toolbox/strings.py

```diff
@@ -102,6 +110,76 @@ def is_invite(string: str, *, fullmatch: bool = True) -> bool:
     return False
 
 
+class MarkdownFormat(IntFlag):
+    NONE = 0
+    STRIKETHROUGH = 1
+    ITALIC_UNDERSCORE = 2
+    ITALIC_ASTERISK = 4
+    BOLD = 8
+    UNDERLINE = 16
+    CODE_BLOCK = 32
+    MULTI_CODE_BLOCK = 64
+    QUOTE = 128
+    MULTI_QUOTE = 256
+    SPOILER = 512
+    ALL = (
+        STRIKETHROUGH
+        | ITALIC_UNDERSCORE
+        | ITALIC_ASTERISK
+        | BOLD
+        | UNDERLINE
+        | CODE_BLOCK
+        | MULTI_CODE_BLOCK
+        | QUOTE
+        | MULTI_QUOTE
+        | SPOILER
+    )
+
+
+format_dict = {
+    MarkdownFormat.MULTI_QUOTE: [r"\s*\>>> ([\s\S]+?)", ">>> {0}"],
+    MarkdownFormat.QUOTE: [r"\s*\> ([\s\S]+?)", "> {0}"],
+    MarkdownFormat.MULTI_CODE_BLOCK: [r"`{3}([\S\s]+?)`{3}", "```{0}```"],
+    MarkdownFormat.CODE_BLOCK: [r"`([^`]+?)`", "`{0}`"],
+    MarkdownFormat.BOLD: [r"\*{2}([\s\S]+?)\*{2}", "\*\*{0}\*\*"],
+    MarkdownFormat.UNDERLINE: [r"__([\s\S]+?)__", "__{0}__"],
+    MarkdownFormat.STRIKETHROUGH: [r"~~([\S\s]+?)~~", "~~{0}~~"],
+    MarkdownFormat.ITALIC_UNDERSCORE: [r"_([^_]+?)_", "_{0}_"],
+    MarkdownFormat.ITALIC_ASTERISK: [r"\*([^*]+?)\*", "\*{0}\*"],
+    MarkdownFormat.SPOILER: [r"\|{2}([\s\S]+?)\|{2}", "\|\|{0}\|\|"],
+}
```

Comment:
Since there are always two values that mean the same thing, I would consider using a `collections.NamedTuple`

Reply:
While I think a NamedTuple is unnecessary here, I do agree a tuple would be better than a list.

Reply:
Done

Reply:
Since there's no named tuple, I think a comment explaining what each index in the tuple is would help readability.